### ISSUE TITLE
Update to Android SDK 12.1.1 for Android 12+ behavior changes

### ISSUE THREAD

**Issue by mabdurrahman** [deepak786/intercom_flutter#230]
Hey folks, we had an issue with Intercom Android SDK and was able to identify the cause, hence, we contacted Intercom support as below to see if they can push a fix or not:
```
Hey Intercom Team, we. recently noticed in our Android application that intercom push notification doesn't open the application or do anything when it's clicked, with further more investigations we discovered it's only affecting Android 12 users and we found some related changes in Android 12 platform which causes the issue: https://developer.android.com/about/versions/12/behavior-changes-12#notification-trampolines

As a workaround right now we downgraded the targetSDKLevel to 30 instead of 31, however, we believe that the full fix should come from Intercom Android SDK side so that they conform to the behavioral changes stated by the guide shared above.

Please let's know whether the Intercom Engineering team is aware already about that issue or not? and whether we have any sooner plans to fix that issue or not?
```
they initially replied:
```
Thank you for flagging this one and going the extra mile to investigate what could have caused the issue 🙌

We are aware of this issue occurring for several users recently, and I can assure you that the team are doing what they can to work on a solution for this issue. I'll keep you updated as soon as I have news from the team about their progress 🤞 In the meantime, I'd recommend that you stay on the downgraded targetSDKLevel. We really appreciate your patience while this issue is being solved 🙏
```
and they recently replied:
```
The team has shipped a new fix in V12.1.1 that will fix the notification issue you experienced 😁 Once you've updated to the newest SDK, you should be good to go 🙌
```

So I think we need to upgrade the version of Intercom Android SDK to allow whoever using targetSDKLevel = 31+ to have their notifications actions working properly as expected

**Comment by deepak786** [comment]
hey @mabdurrahman 

new version `intercom_flutter: ^7.1.0` is released which uses SDK v12.1.1.

Don't hesitate to buy coffee for me
https://www.buymeacoffee.com/deepakdroid

**Comment by mabdurrahman** [comment]
Thanks @deepak786 very much for taking care of this ❤️
